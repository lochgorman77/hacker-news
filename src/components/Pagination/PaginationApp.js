import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Pagination from '@material-ui/lab/Pagination';

import { getPageCount } from 'reducers/stories';

const styles = (theme) => ({
  pagination: {
    marginLeft: 'auto',
    float: 'right',
  },
});

export class PaginationApp extends Component {
  render() {
    return (
      <div id="page" className={this.props.classes.pagination}>
        <Pagination count={this.props.count} page={this.props.page} onChange={this.handleChange} />
      </div>
    );
  }
}

PaginationApp.propTypes = {
  page: PropTypes.number,
  count: PropTypes.number,
  classes: PropTypes.object,
};

const mapStateToProps = (state) => ({
  page: 1,
  count: getPageCount(state),
});

const mapDispatchToProps = (dispatch) => ({});

export const PaginationAppStyle = withStyles(styles, { name: 'PaginationApp' })(PaginationApp);
const ConnectedPaginationAppContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(PaginationAppStyle);

export default ConnectedPaginationAppContainer;
