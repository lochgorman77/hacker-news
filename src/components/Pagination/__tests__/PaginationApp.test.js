import React from 'react';
import { PaginationAppStyle } from '../PaginationApp';

const props = {
  count: 60,
  currentPage: 1,
  handleUpdateStoriesPage: jest.fn(),
  handleResetDetails: jest.fn(),
};

describe('<TopStoriesContainer />', () => {
  let shallow, wrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<PaginationAppStyle {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleChange ', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().handleChange({}, 2);
    expect(spy).toHaveBeenCalled();
    expect(props.handleUpdateStoriesPage.mock.calls.length).toBe(1);
    expect(props.handleResetDetails.mock.calls.length).toBe(1);
  });
});
