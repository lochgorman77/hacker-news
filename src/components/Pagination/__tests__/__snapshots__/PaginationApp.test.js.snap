// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopStoriesContainer /> renders correctly 1`] = `
<div
  className="PaginationApp-pagination"
  id="page"
>
  <MuiPagination
    count={60}
    onChange={[Function]}
    page={1}
  />
</div>
`;
